testgridshot

```diff
@@ -0,0 +1,268 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#+
+#+ Usage: testgridshot <release>
+#+
+#+   <release> can be something like 'master', '1.15', ...
+#+
+#+   This will inspect the testgrid dashboards 'blocking' & 'informing' for the
+#+   specified <release>, create screenshots of failing tests, and print a markdown
+#+   stub on standard out. This stub is intended to be copy & pasted into a issue
+#+   comment for the release cutting GitHub issue.
+#+
+#+ Configuration:
+#+   Configuration can be passed in via the environmnet. The following
+#+   variables are available:
+#+
+#+   BOARDS: [default: "blocking informing"]
+#+     Change which boards you are interested in
+#+
+#+   STATES: [default: "FAILING"]
+#+     Change the tests you want to screenshot
+#+     Available states: FAILING, FLAKY, PASSING
+#+
+#+   BLOCK_WIDTH: [default: 30]
+#+     The width of the individual testgrid red & green block.
+#+     Influences on how many test runs will be shown on the screenshot.
+#+
+#+   WIDTH/HEIGHT: [default: WIDTH=3000/HEIGHT=2500]
+#+     The width and height of the generated screenshot in pixel.
+#+
+#+   RETRY_COUNT: [default: 3]
+#+     Change how often we should retry when talking to (curl'ing) APIs.
+#+
+#+   RETRY_SLEEP: [default: 2]
+#+     Change the amount of time we want to wait between consecutive API call
+#+     retries.
+#+
+#+   Example:
+#+     $ BOARDS='all informing' STATES='FAILING FLAKY' BLOCK_WIDTH=10 ./testgridshot 1.15
+#+         Creates dense screenshots of all the 'sig-release-1.15-{all,informing}'
+#+         boards which are either failing or flaking
+#+
+#+ How it works:
+#+   General flow:
+#+   - Get the dasbhoard URLs we are interested in from testgrid's 'summary'
+#+     endpoint
+#+   - Use 'render-tron.appspot.com' to create screenshots of the testgrid
+#+     boards
+#+   - Upload all the screenshots to 'vgy.me'
+#+   - Print a markdown, which links to the images on 'vgy.me', stub on StdOut
+#+   Remarks:
+#+   - Both 'vgy.me' and 'render-tron.appspot.com' might go away anytime
+#+   - The images are not served by 'vgy.me' directly if used in a GitHub issue,
+#+     but by GitHub's camo service
+#+
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+readonly STATE_FAILING='FAILING'
+readonly STATE_FALKING='FLAKY'
```

Comment:
```suggestion
readonly STATE_FLAKING='FLAKY'
```

Reply:
This appears unused. Do we want to leverage it somewhere or should we remove it?

Reply:
We can remove those, they are not used. I will add a comment about the different states the testgrid summary API returns (of that's not already there) and will remove those.